packages/protocol/test/governance/downtimeslasher.ts

```diff
@@ -0,0 +1,797 @@
+    const bitmapWithoutValidator: string[] = [bitmapVI1, bitmapVI0]
+
+    async function ensureValidatorIsSlashable(
+      startBlock: number,
+      validatorIndices: number[]
+    ): Promise<{ startBlocks: number[]; endBlocks: number[] }> {
+      const bitmapMasks = validatorIndices.map((vi) => bitmapWithoutValidator[vi])
+      await presetParentSealForBlocks(startBlock, slashableDowntime, bitmapMasks)
+      // Signs the outer limits to be 100% secure that the boundaries are well tested
+      await presetParentSealForBlocks(startBlock - 1, 1, [bitmapVI01])
+      await presetParentSealForBlocks(startBlock + slashableDowntime, 1, [bitmapVI01])
+
+      return calculateEverySlot(startBlock)
+    }
+
+    // This function will wait until the middle of a new epoch is reached.
+    // We consider blocks are "safe" if the test could perform a slash, without
+    // the context of the other tests.
+    // This property ensures that each test has an epoch to work with (not depends
+    // on previous tests), and there are enough blocks on that epoch to have a validator
+    // down for a possible slash.
+    async function waitUntilSafeBlocks(safeEpoch: number) {
+      let blockNumber: number = 0
+      const blockStableBetweenTests = getFirstBlockNumberForEpoch(safeEpoch, epochSize) - 1
+      do {
+        blockNumber = await web3.eth.getBlockNumber()
+        await jsonRpc(web3, 'evm_mine', [])
+      } while (
+        blockNumber < blockStableBetweenTests ||
+        // Middle of the epoch
+        blockNumber % epochSize <= epochSize * 0.5
+      )
+    }
+
+    before(async () => {
+      const blockNumber: number = await web3.eth.getBlockNumber()
+      epoch = getEpochNumberOfBlock(blockNumber, epochSize)
+    })
+
+    beforeEach(async () => {
+      await waitUntilSafeBlocks(epoch)
+      await slasher.setNumberValidators(2)
+    })
+
+    afterEach(async () => {
+      const blockNumber: number = await web3.eth.getBlockNumber()
+      const newEpoch = getEpochNumberOfBlock(blockNumber, epochSize)
+
+      // Optimization to make the testing batch run faster.
+      // This "recovers" a gap of more that 1 epoch, and avoids waiting more than an epoch
+      // between tests.
+      if (newEpoch === epoch) {
+        epoch += 1
+      } else {
+        epoch = newEpoch
+      }
+    })
+
+    it('fails if the slash window has the currentBlock as part of the interval', async () => {
+      await slasher.setEpochSigner(epoch, validatorIndexInEpoch, validatorList[0])
+      const actualBlockNumber = await web3.eth.getBlockNumber()
+      const startBlock = actualBlockNumber - 3
+      await assertRevert(
+        slasher.slash(
+          [startBlock, startBlock + intervalSize],
+          [startBlock + intervalSize - 1, startBlock + 2 * intervalSize - 1],
+          [validatorIndexInEpoch],
+          0,
+          [],
+          [],
+          [],
+          [],
+          [],
+          []
+        )
```

Comment:
Could you add a message to check for here?

Reply:
I've changed this and added more tests for this specific fail